### PR TITLE
Ignore the private key handle cert property for persisted keys

### LINKED ISSUE #36273: Issue when accessing the private key from a certificate after it has been exported
This happens when calling GetRSAPrivateKey on a certificate with a persisted private key… after having called Export(PFX) on the cert.
* Cert is found from the store, it has the named private key info property (2), but not the “I have a loaded NCrypt key handle” property (78)

* Call export(PFX).  Inside PFXExportCertStore it needs to load the private key (to export it), and apparently it caches it back on the cert object (property 78).

* Then you call GetRSAPrivateKey().  We see that prop 78 was set, which hitherto has only happened from loading certs in an ephemeral context. So we tell the CngKey object that IsEphemeral is true.

* For… some… reason… the IsEphemeral property writes the property persisted. Since this is actually a persisted key object, the property is actually written to the key store vs just in the ephemeral key’s ephemeral property bag.

* This object is now already loaded on the cert (prop 78) and knows it’s ephemeral (from the property), so it’s happy for all further uses on this instance.

* Read the cert from the store again (same process, different process, doesn’t matter). Prop 2 is set, Prop 78 is not.

* Call GetRSAPrivateKey.  We’re doing a cold-load (no prop 78). We load the key, and check “it better not claim to have a ‘CLR IsEphemeral’ property, because I don’t know what it means for a persisted key to claim it’s ephemeral”.  Explode.

I don’t know that this would have been something that changed in Windows (since I reproed it on 2012R2). We can forward fix the problem by checking for prop 2 before we assert that a prop-78 load means an ephemeral key.
After calling cert2.Export(PFX):

Attached repro can be found here https://github.com/javiercn/certificaterepro

### PR DESCRIPTION
**Pull request by bartonjs**
When a CERT_CONTEXT value has both property 2 (prov info) and property 78
(ncrypt key handle), prefer to load based on the property 2 state.
This avoids a scenario where calling Get[Algorithm]PrivateKey sets the
'CLR IsEphemeral' property on a persisted key, preventing future loads of that key.

An alternative approach of preferring the loaded key over the cold-load was
not selected to avoid value contamination of ephemeral properties set on the
CngKey object directly after the caller calls Get[Algorithm]PrivateKey.

Fixes #36273.

### CODE REVIEW

File: src/libraries/System.Security.Cryptography.X509Certificates/tests/Cert.cs [hunk 1]
```diff
@@ -86,5 +86,10 @@ public void Dispose()
         }
 
         private X509Certificate2[] _certs;
+
+        public static implicit operator ImportedCollection(X509Certificate2Collection collection)
```
Comment (stephentoub) [inline src/libraries/System.Security.Cryptography.X509Certificates/tests/Cert.cs:90]: I'm surprised to see this coming from you :)

Is it really necessary to add this implicit cast?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Security.Cryptography.X509Certificates/tests/Cert.cs (hunk 1 above)
Comment (bartonjs) [reply src/libraries/System.Security.Cryptography.X509Certificates/tests/Cert.cs:90]: > Is it really necessary to add this implicit cast?

Nah.  Added the extra words and removed it.

(Probably what I wanted was to put an accelerator for Find on the wrapper class, but instead I just added the ctor-wraps.)

**Comment by eerhardt** [comment]
allConfigurations failure is #36285

**Comment by bartonjs** [comment]
The only outerloop failure was https://github.com/dotnet/runtime/issues/36207.

**Pull request merged**